File: Practica5/Marco teorico/3. Destructor/E. Destructores y el uso de abort y Exit/3.5.cpp
#include <iostream>
#include <cstdlib> // Para abort() y exit()


using namespace std;


class MiClase {
public:
    MiClase() {
        cout << "Constructor llamado" << endl;
    }


    ~MiClase() {
        // Simulando una condición de error
        bool error = true;


        if (error) {
            cout << "Error detectado en el destructor" << endl;
            // Abortar el programa abruptamente
            abort();
        } else {
            cout << "Destructor llamado normalmente" << endl;
        }
    }
};

int main() {
    cout << "Inicio del programa" << endl;


    MiClase objeto;


    // Forzar una condición de error
    bool error = true;


    if (error) {
        cout << "Error detectado en main()" << endl;
        // Salir del programa con un código de error
        exit(EXIT_FAILURE);
    }


    cout << "Fin del programa" << endl;


    return 0;
}

File: Practica5/Marco teorico/3. Destructor/E. Destructores y el uso de abort y Exit/3.5.py
class MiClase:
    def __init__(self):
        print("Constructor llamado")

    def __del__(self):
        # Simulando una condición de error
        error = True

        if error:
            print("Error detectado en el destructor")
            # Abortar el programa abruptamente
            import sys
            sys.exit(1)
        else:
            print("Destructor llamado normalmente")


print("Inicio del programa")

objeto = MiClase()

error = True

if error:
    print("Error detectado en main()")
    import sys
    sys.exit(1)

print("Fin del programa")
